src/components/ChainConnection.tsx

```diff
@@ -173,23 +173,25 @@ const ChainConnection = () => {
     let connection;
     setConnectionInProgress(true);
     try {
-      const { rpcAddrs, chainName } = await loadNetworkConfig(
-        networkConfig.url
-      );
-      const rpc = sample(rpcAddrs);
+      const config = await loadNetworkConfig(networkConfig.url);
+      const rpc = sample(config.rpcAddrs);
       if (!rpc) {
         throw new Error(Errors.networkConfig);
       }
-      const watcher = makeAgoricChainStorageWatcher(rpc, chainName, e => {
+      const chainId = config.chainName;
+      const watcher = makeAgoricChainStorageWatcher(rpc, chainId, e => {
         console.error(e);
         throw e;
       });
       connection = await makeAgoricWalletConnection(watcher);
       setChainConnection({
         ...connection,
         watcher,
-        chainId: chainName,
+        chainId,
       });
+      for (const message of activeNotices(config)) {
+        toast.warn(message, { position: 'top-center' });
```

Comment:
Do we need to support multiple concurrent notices? It would be be cleaner functionally and aesthetically to reuse https://github.com/Agoric/dapp-psm/blob/main/src/components/AnnouncementBanner.tsx and put this logic in there. If there's multiple maybe we can rethink the design, because having a bunch of toasts pop up isn't ideal either. Not sure the priority of this though, if we have time to think it through more.

Reply:
Good question. I figured there might be one or two max. And I didn't want to risk having someone defined multiple messages and one not go out due to client implemenation. (We don't have any validation tools on the production side.)

I think AnnouncementBanner makes sense for announcements, which can be delivered one a time and dismissed. "Ok, I saw it, got it." Whereas "notice" is more aggressive like, "DO NOT FORGET

Reply:
The announcement banner is what we've used for notices such as chain upgrades in the past. For multiple concurrent notices, I'm thinking it's sufficiently uncommon that showing multiple banners, or multiple rows in the banner, would be fine in that case. I think it should also show when the dapp loads, before the user has to connect their wallet. By doing it that way, the PR won't reinvent any of our existing UX, just make it more convenient to update the notice.